sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -90,7 +90,8 @@ public static int calculateBitSetWidthInBytes(int numFields) {
           FloatType,
           DoubleType,
           DateType,
-          TimestampType
+          TimestampType,
+          TimeType
```

Comment:
Problem is that you have to define the mapping for PySpark and SparkR too that's probably one of the tricky part.
I think it's best to loop the dev mailing list and see if there's a big support for this. Otherwise, I'm not super supportive of it due to the maintenance cost.